syntax = "proto3";

package faiss;

option java_package = "ru.nms.diplom.faiss";
option java_multiple_files = true;

service FaissSearchService {
  rpc search (FaissSearchRequest) returns (FaissSearchResponse);
  rpc encodeQuery (EncodeRequest) returns (EncodedQuery);
}

message FaissSearchRequest {
  repeated float query = 1;
  int32 k = 2;
  bool normalized = 3;
}

message FaissResult {
  int32 id = 1;
  float score = 2;
}

message FaissSearchResponse {
  repeated FaissResult results = 1;
}

// === NEW FOR SIMILARITY ===

message EncodeRequest {
  string query = 1;
}

message EncodedQuery {
  Vector vector = 1;
}

message Vector {
  repeated float values = 1;
}
